scipy/spatial/transform/rotation.py

```diff
@@ -1510,6 +1510,113 @@ def random(cls, num=None, random_state=None):
+            is False.
+
+        Returns
+        -------
+        estimated_rotation : `Rotation` instance
+            Best estimate of the rotation that transforms `b` to `a`.
+        sensitivity_matrix : `numpy.ndarray`, shape (3, 3)
+            Scaled covariance of the attitude errors expressed as the small
+            rotation vector of frame A. Multiply with harmonic mean [2]_ of
+            variance in each observation to get true covariance matrix. The
+            error model is detailed in [3]_.
+
+        References
+        ----------
+        .. [1] F. Landis Markley,
+                “Attitude determination using vector observations: a fast
+                optimal matrix algorithm,” Journal of Astronautical Sciences,
+                Vol. 41, No.2, 1993, pp. 261-280.
+        .. [2] `Harmonic Mean <https://en.wikipedia.org/wiki/Harmonic_mean>`_
+        .. [3] F. Landis Markley,
+                "Attitude determination using vector observations and the
+                Singular Value Decomposition, " Journal of Astronautical
+                Sciences, Vol. 38, No.3, 1988, pp. 245-258.
+        """
+        a = np.asarray(a)
+        if a.ndim != 2 or a.shape[-1] != 3:
+            raise ValueError("Expected input `a` to have shape (N, 3), "
+                             "got {}".format(a.shape))
+        b = np.asarray(b)
+        if b.ndim != 2 or b.shape[-1] != 3:
+            raise ValueError("Expected input `b` to have shape (N, 3), "
+                             "got {}.".format(b.shape))
+
+        if a.shape != b.shape:
+            raise ValueError("Expected inputs `a` and `b` to have same shapes"
+                             ", got {} and {} respectively.".format(
+                                a.shape, b.shape))
+
+        if b.shape[0] == 1:
+            raise ValueError("Rotation cannot be estimated using a single "
+                             "vector.")
+
+        if weights is None:
+            weights = np.ones(b.shape[0])
+        else:
+            weights = np.asarray(weights)
+            if weights.ndim != 1:
+                raise ValueError("Expected `weights` to be 1 dimensional, got "
+                                 "shape {}.".format(weights.shape))
+            if weights.shape[0] != b.shape[0]:
+                raise ValueError("Expected `weights` to have number of values "
+                                 "equal to number of input vectors, got "
+                                 "{} values and {} vectors.".format(
+                                    weights.shape[0], b.shape[0]))
+        weights = weights / np.sum(weights)
+
+        if not normalized:
+            a = a / scipy.linalg.norm(a, axis=1)[:, None]
+            b = b / scipy.linalg.norm(b, axis=1)[:, None]
+
+        B = np.einsum('ji,jk->ik', weights[:, None] * a, b)
+        u, s, vh = np.linalg.svd(B)
+        C = np.dot(u, vh)
+
+        zeta = (s[0]+s[1]) * (s[1]+s[2]) * (s[2]+s[0])
+        if zeta == 0:
```

Comment:
Any progress with zeta threshold?

Reply:
Could you elaborate a little why you came to the conclusion that `zeta == 0` is a proper threshold?